### PR TITLE
Adding missing public docstrings to Bio.Phylo, Issue 1203

### PR DESCRIPTION
**Pull request by svalqui**
This pull request addresses issue #1203 

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/Phylo/PhyloXML.py [hunk 1]
```diff
@@ -389,6 +389,7 @@ def to_phylogeny(self, **kwargs):
     # Shortcuts for list attributes that are usually only 1 item
     # NB: Duplicated from Phylogeny class
     def _get_confidence(self):
+        """Return confident values (PRIVATE)."""
```
Comment (peterjc) [inline Bio/Phylo/PhyloXML.py]: confident --> confidence

File: Bio/Phylo/PhyloXML.py [hunk 2]
```diff
@@ -668,27 +683,45 @@ def __floordiv__(self, other):
...
+        """Return quotient and remainder.
+
+        of dividing the values of two Confidence objects.
+        """
         return divmod(self.value, other)
 
     def __rdivmod__(self, other):
+        """Return quotient and remainder.
+
+        of dividing and reversing the values of two Confidence objects.
```
Comment (peterjc) [inline Bio/Phylo/PhyloXML.py]: This isn't a self-contained sentence, and I'm unclear on your meaning.

File: Bio/Phylo/PhyloXML.py [hunk 3]
```diff
@@ -668,27 +683,45 @@ def __floordiv__(self, other):
...
+        """
         return divmod(other, self.value)
 
     def __pow__(self, other, modulo=None):
+        """Return the value of Confident object raised at power of.
```
Comment (peterjc) [inline Bio/Phylo/PhyloXML.py]: This does not make sense (unless you include the rest of the docstring). How about:

```
"""Return value of confidence object raised to the given power."""

Also, typo: Confident object.

File: Bio/Phylo/PhyloXML.py [hunk 4]
```diff
@@ -668,27 +683,45 @@ def __floordiv__(self, other):
...
+
+        the value of other Confident object.
+        """
         if modulo is not None:
             return pow(self.value, other, modulo)
         return pow(self.value, other)
 
     def __rpow__(self, other):
+        """Return the value of other Confident object raised at power of.
```
Comment (peterjc) [inline Bio/Phylo/PhyloXML.py]: This doesn't make sense either, see ``__pow__`` comment.

File: Bio/Phylo/PhyloXML.py [hunk 5]
```diff
@@ -1384,8 +1429,7 @@ def __init__(self, value, desc=None, type=None):
         self.type = type
 
     def __str__(self):
+        """Return string representation of Uri."""
         if self.value:
             return self.value
-        """Initialize the class."""
-        """Initialize the class."""
```
Comment (peterjc) [inline Bio/Phylo/PhyloXML.py]: Those look like an accidental paste - well spotted. I wonder if there is a flake8 setting to report this kind of thing?

File: Bio/Phylo/PhyloXMLIO.py [hunk 6]
```diff
@@ -488,6 +495,7 @@ def _parse_taxonomy(self, parent):
         return taxonomy
 
     def other(self, elem, namespace, localtag):
+        """Set non-phyloXML element."""
```
Comment (peterjc) [inline Bio/Phylo/PhyloXMLIO.py]: Is "set" the right word here? In the context of Python properties that verb is used to assign a new value.

(Same comment applies to the following methods too)

**Comment by svalqui** [comment]
@peterjc , Thank you for the feedback, I think is better now, what do you think.

**Comment by peterjc** [comment]
What do you think @etal? Is this ready to merge?

**Comment by peterjc** [comment]
Looks like Eric has been too busy - I'm going to merge this anyway as I think it is an good improvement, even if with Eric's help we could make this even better.

**Comment by peterjc** [comment]
Thank you @svalqui 👍

**Pull request merged**